index.d.ts

```diff
@@ -0,0 +1,356 @@
+// Type definitions for [~THE LIBRARY NAME~] [~OPTIONAL VERSION NUMBER~]
+// Project: [~THE PROJECT NAME~]
+// Definitions by: [~YOUR NAME~] <[~A URL FOR YOU~]>
+
+export as namespace Draggable;
+
+export type DraggableContainer = HTMLElement | HTMLElement[] | NodeList
+
+declare enum DraggableEventName {
+  Initialized = 'draggable:initialized',
+  Destroy = 'draggable:destroy'
+}
+
+declare enum DragEventName {
+  Start = 'drag:start',
+  Move = 'drag:move',
+  Over = 'drag:over',
+  OverContainer = 'drag:over:container',
+  Out = 'drag:out',
+  OutContainer = 'drag:out:container',
+  Stop = 'drag:stop',
+  Pressure = 'drag:pressure'
+}
+
+declare enum DropEventName {
+  Over = 'droppable:over',
+  Out = 'droppable:out'
+}
+
+declare enum SortEventName {
+  Start = 'sortable:start',
+  Sort = 'sortable:sort',
+  Sorted = 'sortable:sorted',
+  Stop = 'sortable:stop'
+}
+
+declare enum SwapEventName {
+  Start = 'swappable:start',
+  Swap = 'swappable:swap',
+  Swapped = 'swappable:swapped',
+  Stop = 'swappable:stop'
+}
+
+declare enum MirrorEventName {
+  Created = 'mirror:created',
+  Attached = 'mirror:attached',
+  Move = 'mirror:move',
+  Destroy = 'mirror:destroy'
+}
+
+export class BasePlugin {
+  draggable: Draggable
+
+  constructor(draggable: Draggable)
+
+  attach(): void
+
+  detach(): void
+}
+
+export type AbstractPlugin = typeof BasePlugin
```

Comment:
FYI, added an `AbstractPlugin` class with the release of `beta.4`: https://github.com/Shopify/draggable/blob/master/src/shared/AbstractPlugin/AbstractPlugin.js

Reply:
@tsov here the type statement AbstractPlugin has to be ``type`` because the ``AbstractPlugin `` has been re-export as ``BasePlugin`` in entry of Draggable module.
https://github.com/Shopify/draggable/blob/5ff60fd7232c82a1546b803028fa1247501cbed9/src/index.js#L14

So I just declare and export ``BasePlugin`` as ``AbstractPlugin`` shape and also export its type reference.

Reply:
@tsov thx, I have changed the type declaration of  ``AbstractPlugin `` and re-export it as ``BasePlugin``. 

see in haoliangwu@588408c.